### DISCUSSION TITLE
Left join random table based on data

### DISCUSSION THREAD

**Discussion by webnoob** [prisma/prisma#4437] [Q&A] [answered]
Is this query possible without using `queryRaw`?

```sql
select
a.id as arcId,
b.id as bookId, 
n.*,
qi.*
from 
notificationqueueitem as qi
inner join notificationqueue as q on q.id = qi.queueId
inner join notification as n on n.id = q.notificationId
left join arc as a on q.entityId = a.id and q.entityType = 'arc' <---- THIS
left join book as b on q.entityId = b.id and q.entityType = 'book' <---- AND THIS
where
qi.userId = 3;
```

Basically, I want to bring back a model depending on the entity type in this column.

I'm debating adding an optional relationship to the model to get around it but thought I'd check first.

**Comment by webnoob** [comment, marked as answer]
I've decided to just add the optional tables against the model. Seems cleaner as I can just include them and then check for null.

**Reply by ryands17** [reply]
Hey @webnoob 👋 
A random table meaning it isn't connected via relations? If so, then you would need to use the above workaround that you have specified :)

**Reply by webnoob** [reply]
Sorry the schema would have helped.

The entityType is another model in the schema, the entityId is the id of that model.

Normally, I'd do `findUnique({ include: arc: true })`.

So the schema would be:

```prisma
model NotificationQueue {
  id Int  @default(autoincrement()) @id

  scheduledDate DateTime @default(now())
  status Int
  statusDate DateTime

  entityType String
  entityId Int
}

model Arc {
  id Int
}

model Book {
  id Int
}
```

I would want to optionally bring back the Arc OR the Book model.

**Reply by ryands17** [reply]
Yeah, as the models are not connected via relations, Prisma wouldn't be able to fetch them, so either a Raw query or adding some sort of relation against the model would work.

**Reply by webnoob** [reply]
Ok thanks for confirming. After I posted it seemed that would be the only way so I've changed my model to include a 1 to many relationship instead. I can include them all and just check the code for null that way.

Thanks for your time once again.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
